lib/internal/util.js

```diff
@@ -797,6 +798,37 @@ function guessHandleType(fd) {
   return handleTypes[type];
 }
 
+class WeakReference {
+  constructor(object) {
+    this.weak = new SafeWeakRef(object);
```

Comment:
Drive-by comment: this [will fail](https://github.com/tc39/ecma262/pull/1307) if it's instantiated in a realm where someone has run `Object.defineProperty(Object.prototype, 'weak', { writable: false })`.

I'm not familiar enough with node internals to know if that's actually possible here, though the use of `__proto__: null` within this file suggests it might be.

Unfortunately TC39 hasn't yet gotten around to [fixing](https://github.com/tc39/ecma262/issues/1036) `extends null` (it's in my queue!), so the easiest fix for this is to use class fields. Putting `weak; strong; refCount;` in the class body will cause the fields to be installed on the instance with `defineProperty`, which bypasses the "override mistake". Using private fields instead will also avoid the problem, of course.

Reply:
Thanks for catching it, updated to use private fields (I was already thinking about that but then this was entirely internal so I thought, might as well save typing a `#`..)